raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -131,20 +131,36 @@ function makeAndSignTransfer$(
     'expiration too soon',
   );
 
-  // fee is added to the lock amount, check for overflow
-  const amountWithFee = decode(
-    UInt(32),
-    action.payload.value.add(fee),
-    'overflow of locked amount with fee',
-  );
+  // fee is added to the lock amount; overflow is checked on locksSum below
+  const amountWithFee = action.payload.value.add(fee) as UInt<32>;
+  const expiration = BigNumber.from(
+    action.payload.expiration ||
+      Math.min(
+        state.blockNumber + revealTimeout * 2,
+        state.blockNumber + channel.settleTimeout - confirmationBlocks,
```

Comment:
Don't we enforce that `settleTimeout >= 2 * revealTimeout`? So in the edge case we deduct the confirmation window...
I can't really pinpoint, but this looks strange

Reply:
On LC, at least, we don't. `revealTimeout` is expected to be smaller than `settleTimeout`, but we aren't sure of it either. On LC, `revealTimeout` is a config global value, while `settleTimeout` comes from the blockchain and can be as low as 20 blocks. `expiration` can come from the `transfer/request` action (e.g. in case of a mediated transfer), and therefore is validated below, but if not provided, we need this just as a default value. Of course, it needs to be larger than `revealTimeout` (which defines the dangerZone), and this check is also done both here and on receiving, but there's no actual protocol/security requirement how large it needs to be, there need to be just enough _time_ for a transfer to succeed, therefore this check here ensures we chose the minimum between available expirations, and let the expiration check fail if it isn't fit. Some tests include revealTimeout=50 and settleTimeout=60, and it works with this code.

Reply:
But why do we need to subtract `confirmationBlocks`? Shouldn't `state.blockNumber + channel.settleTimeout` be valid as well?

Reply:
We need to give **some time** for the transfer to succeed. We give confirmationBlocks here, only as a default (in case `2 * revealTimeout < settleTimeout`)

Reply:
Isn't `revealTimeout` exactly the time we assume a tx will need (maximally) to get mined?

Reply:
It is, for transactions. We need to give some time before this (danger zone) for the off-chain protocol to work.

Reply:
Ok. now I get it. Thanks for talking me through!